### PR TITLE
SUBMARINE-481. Use Swagger to submarine RESTful APIs

### PR DESCRIPTION
**Pull request by pingsutw**
### What is this PR for?
Use swagger annotation to generate API document 
https://gist.github.com/pingsutw/4d96afb54d4c767304f74844b432585c
1. Build Submarine source code
2. start submarine server
3. open localhost:8080/v2/swagger.json

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-481

### How should this be tested?
https://travis-ci.org/github/pingsutw/hadoop-submarine/builds/680491536

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-server/server-core/pom.xml [hunk 1]
```diff
@@ -38,6 +38,46 @@
       <version>${project.version}</version>
     </dependency>
 
+    <dependency>
+      <groupId>io.swagger</groupId>
+      <artifactId>swagger-jersey2-jaxrs</artifactId>
+      <version>1.5.18</version>
```
Comment (jiwq) [inline submarine-server/server-core/pom.xml:44]: Should define the version in properties.

File: submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml [hunk 2]
```diff
@@ -28,10 +28,34 @@
...
+  </servlet>
+  <servlet>
+    <servlet-name>Bootstrap</servlet-name>
+    <servlet-class>org.apache.submarine.server.Bootstrap</servlet-class>
+    <load-on-startup>2</load-on-startup>
+  </servlet>
+
+  <servlet-mapping>
+    <servlet-name>default</servlet-name>
+    <url-pattern>/v2/*</url-pattern>
```
Comment (jiwq) [inline submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml:56]: why use the `v2`?

Comment (tangzhankun) [reply submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml:56]: do you mean swagger v3?

**Comment by tangzhankun** [comment]
@kevin85421 thanks for the contribution. I'm wondering is it better to start a swagger API document in this PR too?

**Comment by tangzhankun** [comment]
> @kevin85421 thanks for the contribution. I'm wondering is it better to support the swagger UI for API document in this PR too?

Let's enable the swagger UI later.

**Comment by pingsutw** [comment]
Thanks @tangzhankun @jiwq for the review.
> Let's enable the swagger UI later.

yes, it will be great. I'm going to do it.

File: submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml (hunk 2 above)
Comment (pingsutw) [reply submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml:56]: This path is for the user to check swagger.json? like `localhost:8088/v2/swagger.json`
it just an arbitrary name, no special meaning.
Maybe we could change to other name. Any suggestion?

Comment (jiwq) [reply submarine-workbench/workbench-web-ng/src/WEB-INF/web.xml:56]: I think it should keep align with the version of submarine REST api, at now it used `v1`.

**Comment by pingsutw** [comment]
I have attached the experiment.yaml and simple doc about experiments spec in [jira](https://issues.apache.org/jira/browse/SUBMARINE-481), feel free to comment and edit.

**Comment by liuxunorg** [comment]
This PR has not been merged for too long, I suggest merging it in first, if there is a problem, then recreate a new PR to fix.

**Comment by tangzhankun** [comment]
@liuxunorg  yeah. I agree. Let's get this in to enable swagger annotation first.

**Pull request closed without merging**